### PR TITLE
[ONNX] Update ONNX docs for release

### PR DESCRIPTION
**Pull request by neginraoof**
ONNX doc updates.

### CODE REVIEW

File: torch/csrc/onnx/onnx.h [hunk 1]
```diff
@@ -20,5 +20,5 @@ enum class TrainingMode {
 // onnx::IR_VERSION. with this change, the test_operators.py will be more
 // stable. only bump it when it's necessary
 static const size_t IR_VERSION = 6;
-static const char* PRODUCER_VERSION = "1.7";
+static const char* PRODUCER_VERSION = "1.8";
```
Comment (BowenBao) [inline torch/csrc/onnx/onnx.h:23]: does this change not affect expect file tests?

Comment (neginraoof) [reply torch/csrc/onnx/onnx.h:23]: No, test files have a placeholder for this variable.

Comment (neginraoof) [reply torch/csrc/onnx/onnx.h:23]: https://github.com/pytorch/pytorch/blob/1fd48a9d1f4158fb015793caa282d2c0ca92059d/test/onnx/expect/TestOperators.test_acos.expect#L3

**Review by BowenBao — APPROVED** [review]
Thanks for the updates!

**Review by bzinodev — APPROVED** [review]

**Pull request closed without merging**